packages/shylu/shylu_dd/frosch/src/Tools/FROSch_Tools_decl.hpp

```diff
@@ -58,6 +58,14 @@
 #define FROSCH_TIMER_STOP(A) A.reset();
 #endif
 
+#ifndef FROSCH_WARNING
+#define FROSCH_WARNING(CLASS,VERBOSE,OUTPUT) if (VERBOSE) std::cerr << CLASS << " : WARNING: " << OUTPUT << std::endl;
+#endif
+
+#ifndef FROSCH_NOTIFICATION
+#define FROSCH_NOTIFICATION(CLASS,VERBOSE,OUTPUT) if (VERBOSE) std::cout << CLASS << " : NOTIFICATION: " << OUTPUT << std::endl;
+#endif
+
 #ifndef FROSCH_TEST_OUTPUT
 #define FROSCH_TEST_OUTPUT(COMM,VERBOSE,OUTPUT) COMM->barrier(); COMM->barrier(); COMM->barrier(); if (VERBOSE) std::cout << OUTPUT << std::endl;
```

Comment:
If you want to print on all processes, consider using something like `Tpetra::Details::gathervPrint` (see `Trilinos/packages/tpetra/core/src/Tpetra_Details_gathervPrint.hpp`).  That tends to produce more consistent output, even if only a few processes end up printing something.